app/services/email_output_service.rb

```diff
@@ -37,19 +38,26 @@ def send_emails_with_attachments(action, email_attachments, submission_id:)
           current_email: index + 1,
           number_of_emails: email_attachments.size,
           submission_id: submission_id
-        )
+        ),
+        submission_id: submission_id
       )
     end
   end
 
-  def send_single_email(subject:, action:, attachments: [])
-    emailer.send_mail(
-      from: action.fetch(:from),
-      to: action.fetch(:to),
-      subject: subject,
-      body_parts: email_body_parts(action.fetch(:email_body)),
-      attachments: attachments
-    )
+  def send_single_email(subject:, action:, attachments: [], submission_id:)
+    email_payload = find_or_create_email_payload(submission_id, attachments)
```

Comment:
Just curious, is there any possibility that attachments can be mess up in filestore (change names, etc) and this start to create many email payload records?

Reply:
yeah unfortunately the filename is the only unique identifier we have for uploads at the moment :(

we have a story about filenames don't we?

Reply:
Yes, we have. Can we use the email count for the finder? By the way, the PR looks good and good to merge when CI pass. Just trying to avoid problems in the future in live prod 🙈